src/ocean/util/serialize/contiguous/MultiVersionDecorator.d

```diff
@@ -181,6 +314,136 @@ class VersionDecorator
 
         assert(0);
     }
+
+    /// ditto
+    public Contiguous!(S) handleVersion(S)
+        (ref void[] buffer, Version.Type input_version)
+    {
+        return handleVersion!(S)(*cast(Buffer!(void)*) &buffer,
+            input_version);
+    }
+
+    /***************************************************************************
+
+        Helper method that takes care of actual conversion routine between
+        two struct types (those are assumed to be of compatible versions)
+
+        Uses this.convert_buffer for temporary allocations
+
+        Template_Params:
+            S = needed struct type
+            Source = struct type seralized into buffer
+
+        Params:
+            buffer = contains serialized Source instance, will be modified to
+                store deserialized S instance instead.
+
+    ***************************************************************************/
+
+    public Contiguous!(S) convert (S, Source) (ref Buffer!(void) buffer)
```

Comment:
Is this intended to be used by the user? If so, does it missing feature notes? If not, can we make `/* package */` comment for the future use?

Reply:
Also, perhaps add some unittests for this method.

Reply:
Right now it is used from `MapSerializer` - https://github.com/sociomantic-tsunami/ocean/blob/5c01dc0adee97b833054a8cd1efd0ca29c8d8cea/src/ocean/util/container/map/utils/MapSerializer.d#L1181

It is tested as part of generic version decorator tests.

Reply:
In case that was not clear - it is not a new method, and it was already public. Only difference is that it used to come from mixin and now defined directly in the class.